### PR TITLE
VIRA-293: allow access token auth

### LINKED ISSUE #76: Allow Personal Access Token (PAT) Auth
Personal access tokens authenticate via the 'Authorization: Bearer xxx' header. It looks like VIRA only supports basic auth at the moment. 

Description of the feature is here: https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html

When using the python API, these creds can be provided via:

```
jir = JIRA('https://hello.planet.com/jira', options={"headers": {"Authorization": "Bearer xxxx"}})
```

### PR DESCRIPTION
**Pull request by jacobstr**
* Tested access_token and access_token_cmd.
* Have not tested for regressions with username and password.
Fixes #76

### CODE REVIEW

File: python/Vira/vira_api.py [hunk 1]
```diff
@@ -202,16 +213,21 @@ def connect(self, server):
...
-                timeout=2,
-                async_=True,
-                max_retries=2)
+            kwargs={
+                "timeout": 2,
+                "async_": True,
+                "max_retries": 2,
+                "options": option_kwargs,
+            }
+            kwargs.update(auth_kwargs)
```
Comment (jacobstr) [inline python/Vira/vira_api.py:228]: I kind of hate the way way I construct the final kwargs array, but I didn't want to import some deep-dict-merge module for this. So right now it's a bit ratty and probably fails cyclomatic complexity scores with the janky if/else loop earlier.

I don't think it's quite yet shit enough to refactor :P I do think that dictionary.update sucks by default and doesn't do what one expects (sensible recursive merge ala jsonpatch).

**Comment by n0v1c3** [comment]
Sorry I have been "away" for a bit. I am going to take a look at all the new posted issues then a real look into this one.

**Comment by n0v1c3** [comment]
I am currently getting the error:

```python
No `vira_servers.json/yaml` file found or, check the config! See `README.md` for more information.                                      
Error detected while processing function vira#_set[19]..vira#_connect:                                                                  
line   14:                                                                                                                              
Traceback (most recent call last):                                                                                                      
  File "<string>", line 1, in <module>                                                                                                  
  File ".../github/jacobstr/vira/python/Vira/vira_api.py", line 198, in connect                          
    auth_kwargs['basic_auth'] == (username, password)                                                                                   
KeyError: 'basic_auth'
```

I am sure there is a quick fix for it but I am going to look into why it turned backwards on us to try and catch it with the proper `try/catch`

**Comment by n0v1c3** [comment]
I even knew I didn't look close enough at it. The core of the problem for the existing users will be the `==` on line `198` this was supposed to be an `=` for the group you were not as worried about ;-).

Keeping it working would be the nicest I can do the merge and the fix or you can do the fix if you would like. If it still works for both of us I will release it to `master` and do a backwards merge with `dev` and then back into the `master` branch.

**Comment by jacobstr** [comment]
Heh thanks @n0v1c3 - I'm fine if you patch it up since you were able to repro it and will be on deck to validate the fix. I ran through a few scenarios by hand but clearly missed this.

**Comment by n0v1c3** [comment]
@jacobstr I currently have an open branch due to development VIRA-293 should exist until the full merge is complete.

**Comment by mikeboiko** [comment]
I implemented this feature a few months ago.
Can you confirm if it works for you, please?

**Pull request closed without merging**